### ISSUE TITLE
Exception when building a serializer for type with a missing field from the schema

### ISSUE THREAD

**Issue by nicodeslandes** [ch-robinson/dotnet-avro#255]
I'm getting an error when attempting to create a serializer for my schema.

Here's the schema, which include a new field called `Values`, which is an array of records with a default value of `[]`:

```json
{
  "type": "record",
  "name": "Order",
  "namespace": "test",
  "fields": [
    { "name": "Id", "type": "int" },
    {
      "name": "Values",
      "type": {
        "type": "array",
        "items": {
          "type": "record",
          "name": "InnerObject",
          "fields": [
            { "name": "Value", "type": "int" }
          ]
        }
      },
      "default": []
    }
  ]
}
```

Things work fine if I the C# type I use contains the new field:

```c#
class InnerObject
{
    public int Value { get; set; }
}

class Order
{
    public int Id { get; set; }
    public InnerObject[] Values { get; set; } = Array.Empty<InnerObject>();
}

// OK
var serializer = new BinarySerializerBuilder().BuildDelegate<Order>(schema);
```

But if I attempt to generate the serializer with the "old" version of my C# type, ie without the `Values` property, the serializer builder fails with this error:
`System.InvalidOperationException: 'variable '' of type 'System.Collections.Generic.IEnumerator'1[System.Object]' referenced from scope 'InnerObject serializer', but it is not defined'`

```c#

class OrderLegacy
{
    public int Id { get; set; }
}

// FAIL: System.InvalidOperationException
var serializer = new BinarySerializerBuilder().BuildDelegate<Order>(schema);
```

**Comment by dstelljes** [comment]
Thanks again for the fix; will release with 9.5.1 later today.

**Comment by nicodeslandes** [comment]
No problem. Thank you so much for merging that PR so quickly!
